src/PostgREST/DbStructure.hs

```diff
@@ -721,50 +681,52 @@ allPrimaryKeys tabs =
                 WHEN ss.contype = 'f' THEN information_schema._pg_index_position(ss.conindid, ss.confkey[(ss.x).n])
                 ELSE NULL::integer
             END::integer AS position_in_unique_constraint
-        FROM pg_attribute a,
-            ( SELECT r.oid AS roid,
-                r.relname,
-                r.relowner,
-                nc.nspname AS nc_nspname,
-                nr.nspname AS nr_nspname,
-                c.oid AS coid,
-                c.conname,
-                c.contype,
-                c.conindid,
-                c.confkey,
-                information_schema._pg_expandarray(c.conkey) AS x
-               FROM pg_namespace nr,
-                pg_class r,
-                pg_namespace nc,
-                pg_constraint c
-              WHERE
-                nr.oid = r.relnamespace
-                AND r.oid = c.conrelid
-                AND nc.oid = c.connamespace
-                AND c.contype in ('p', 'u', 'f')
-                AND r.relkind IN ('r', 'p')
-                AND NOT pg_is_other_temp_schema(nr.oid)
-            ) ss
+        FROM pg_attribute a
+        JOIN (
+          SELECT r.oid AS roid,
+            r.relname,
+            r.relowner,
+            nc.nspname AS nc_nspname,
+            nr.nspname AS nr_nspname,
+            c.oid AS coid,
+            c.conname,
+            c.contype,
+            c.conindid,
+            c.confkey,
+            information_schema._pg_expandarray(c.conkey) AS x
+          FROM pg_namespace nr
+          JOIN pg_class r
+            ON nr.oid = r.relnamespace
+          JOIN pg_constraint c
+            ON r.oid = c.conrelid
+          JOIN pg_namespace nc
+            ON c.connamespace = nc.oid
+          WHERE
+            c.contype in ('p', 'u')
```

Comment:
Why was `'f'` removed here?

Reply:
@wolfgangwalther I elaborated on that on this commit message: https://github.com/PostgREST/postgrest/pull/2250/commits/37fa43c879f32e5c9351d45a605a6257bbfa0431

This query was meant for getting all the primary key cols and the 'f'(foreign key) doesn't make sense there.

I assumed this happened because this query was copied from `information_schema.key_column_usage` <- which searches all constraints.

'u' should likely be removed as well.